crates/astria-sequencer/src/proposal/commitment.rs

```diff
@@ -22,35 +32,49 @@ impl GeneratedCommitments {
 
 /// Called when we receive a `PrepareProposal` or `ProcessProposal` consensus message.
 ///
-/// In the case of `PrepareProposal`, we use this function to generate the `commitment_tx`
+/// In the case of `PrepareProposal`, we use this function to generate the `rollup_datas_commitment`
 /// to be placed at the start of the block.
 ///
 /// In the case of `ProcessProposal`, we use this function to generate and verify the
-/// `commitment_tx` expected at the start of the block.
+/// `rollup_datas_commitment` expected at the start of the block.
 ///
 /// This function sorts the block's `sequence::Action`s contained within the transactions
-/// using their `rollup_id`. It then returns the merkle root of the tree where each leaf is
-/// a commitment of `sequence::Action`s with the same `rollup_id`. The leaves are ordered
-/// by `rollup_id` in ascending order.
-/// This structure can be referred to as the "action tree".
+/// using their `rollup_id`. It also appends the `Deposit`s generated by the block execution
+/// to each rollup's sequenced data.
+/// It then returns the merkle root of the tree where each leaf is
+/// a commitment of the rollup data (`sequence::Action`s and `Deposit`s) with the same `rollup_id`.
+/// The leaves are ordered by `rollup_id` in ascending order.
 ///
 /// The leaf, which contains a commitment to every action with the same `rollup_id`, is currently
 /// implemented as ( `rollup_id` || root of merkle tree of the `sequence::Action`s ).
-/// This is somewhat arbitrary, but could be useful for proof of an action within the action tree.
-pub(crate) fn generate_sequence_actions_commitment(
+/// This is somewhat arbitrary, but could be useful for proof of an action within the rollup datas
+/// tree.
+pub(crate) fn generate_rollup_datas_commitment(
     signed_txs: &[SignedTransaction],
+    deposits: HashMap<RollupId, Vec<Deposit>>,
 ) -> GeneratedCommitments {
-    let rollup_ids_to_txs = astria_core::sequencer::v1alpha1::group_sequence_actions_in_signed_transaction_transactions_by_rollup_id(signed_txs);
+    use prost::Message as _;
+
+    let mut rollup_ids_to_txs =
+        group_sequence_actions_in_signed_transaction_transactions_by_rollup_id(signed_txs);
     let rollup_ids_root = merkle::Tree::from_leaves(rollup_ids_to_txs.keys()).root();
 
+    for (rollup_id, deposit) in deposits {
```

Comment:
`rollup_ids_to_txs` *must* be srotede after this operation.

Reply:
fixed